src/states/client.rs

```diff
@@ -200,42 +194,21 @@ impl Client {
         pub_id: PublicId,
         outbox: &mut EventBox,
     ) -> Result<Transition> {
-        if self.proxy_pub_id == pub_id {
-            hop_msg.verify(self.proxy_pub_id.signing_public_key())?;
-        } else {
+        if self.proxy_pub_id != pub_id {
             return Err(RoutingError::UnknownConnection(pub_id));
         }
 
-        let signed_msg = hop_msg.content;
-        signed_msg.check_integrity(self.min_section_size())?;
-
-        let routing_msg = signed_msg.into_routing_message();
-        let in_authority = self.in_authority(&routing_msg.dst);
-
-        // Prevents us repeatedly handling identical messages sent by a malicious peer.
-        match self
-            .routing_msg_filter
-            .filter_incoming(&routing_msg, hop_msg.route)
-        {
-            FilteringResult::KnownMessage | FilteringResult::KnownMessageAndRoute => {
-                return Err(RoutingError::FilterCheckFailed);
-            }
-            FilteringResult::NewMessage => (),
-        }
-
-        if !in_authority {
-            return Ok(Transition::Stay);
+        if let Some(routing_msg) = self.filter_hop_message(hop_msg, pub_id, false)? {
```

Comment:
I know what that false is for but it is not very nice.
Could it be a constant implemented by the trait https://doc.rust-lang.org/edition-guide/rust-2018/trait-system/associated-constants.html ? Would it help?

Reply:
:+1: Sounds like a good idea!